analytic_engine/src/compaction/scheduler.rs

```diff
@@ -600,67 +605,28 @@ impl ScheduleWorker {
     }
 
     async fn schedule(&mut self) {
-        self.purge_tables();
+        self.purge_tables().await;
         self.flush_tables().await;
     }
 
-    fn purge_tables(&mut self) {
+    async fn purge_tables(&mut self) {
```

Comment:
It is weird that the purge procedure seems removed? I guess we should add a procedure for periodical compaction rather than replacing the purge procedure with the new compaction procedure.

Reply:
The purge logic is already in `compact_table`, I just reuse  `handle_table_compaction_request` function to do the compaction.